### PR TITLE
Create a URL which initiates a library search

### PR DESCRIPTION
**Pull request by jsnshrmn**
[//]: # (Thank you for uploading a PR to the Wikipedia Library!)

## Description
Adds:
- an intentionally minimal page with  an intermediary form at `/search/?q=<search terms>` that sends users from stable urls on-wiki to searches in EDS.
- an eligible editors only view mixin that raises permission denied for users without an editor and redirects ineligible editors to my_library, which explains eligibility

Updates:
- the referring url policy to be in django instead of nginx. the header was getting added twice and it was causing intermittent problems during testing
- Drops the `from_homepage` query parameter once our auth check is done using it

## Rationale
For use in the Find Sources template

## Phabricator Ticket
https://phabricator.wikimedia.org/T294919

## How Has This Been Tested?
I added unit tests for the new view mixin, since that is reusable and has security implications.
I manually tested:
- multiple locales, including those that need the code manipulated, like `pt`, with JavaScript enabled and disabled.
- eligible and ineligible editor accounts
- anonymous and logged in users

## Screenshots of your changes (if appropriate):
nojs screenshot; the javascript version automatically sends the user to EDS.
image

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Minor change (fix a typo, add a translation tag, add section to README, etc.)

### CODE REVIEW

File: TWLight/urls.py [hunk 1]
```diff
@@ -88,6 +88,11 @@
     url(r"^contact/$", ContactUsView.as_view(), name="contact"),
     url(r"^$", NewHomePageView.as_view(), name="homepage"),
     url(r"^about/$", TemplateView.as_view(template_name="about.html"), name="about"),
+    url(
+        r"^search$",
```
Comment (suecarmol) [inline TWLight/urls.py:92]: Did you leave the `/` out on purpose? Otherwise, I think you should add it

**Review by suecarmol — CHANGES_REQUESTED** [review]
Just a small change before I can merge this

File: TWLight/urls.py (hunk 1 above)
Comment (jsnshrmn) [reply TWLight/urls.py:92]: Yeah, it really is out of alignment with literally every other route. Fixing!

Comment (jsnshrmn) [reply TWLight/urls.py]: Done!

**Review by suecarmol — APPROVED** [review]
Thank you for working on this!

**Pull request merged**